tenant/store.go

```diff
@@ -49,3 +48,149 @@ type Segment struct {
 	Name     string
 	Seq      uint64
 }
+
+func (tenantStore *tenantStore) listTenants() ([]Tenant, error) {
+	var tenants []Tenant
+	log.Println("In listTenants()", &tenants)
+	tenantStore.DbStore.Db.Find(&tenants)
+	err := common.MakeMultiError(tenantStore.DbStore.Db.GetErrors())
+	if err != nil {
+		return nil, err
+	}
+	log.Println(tenants)
+	return tenants, nil
+}
+
+func (tenantStore *tenantStore) listSegments(tenantId uint64) ([]Segment, error) {
+	var segments []Segment
+	log.Println("In listTenants()", &segments)
+	tenantStore.DbStore.Db.Where("tenant_id = ?", tenantId).Find(&segments)
+	err := common.MakeMultiError(tenantStore.DbStore.Db.GetErrors())
+	if err != nil {
+		return nil, err
+	}
+	log.Println(segments)
+	return segments, nil
+}
+
+func (tenantStore *tenantStore) addTenant(tenant *Tenant) error {
+	var tenants []Tenant
+	tenantStore.DbStore.Db.Find(&tenants)
+
+	//	myId := tenant.Id
+	tenant.Seq = uint64(len(tenants) + 1)
```

Comment:
We ignored deletion for the time being... Usually in systems like these I do not like even to ever DELETE, but to set a deletion flag. (Who knows what 'delete' really means - it could be an accidental cancellation of an account).